query_ranges_to_vnodes_async.cc

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2025-present ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: LicenseRef-ScyllaDB-Source-Available-1.0
+ */
+
+#include "query_ranges_to_vnodes_async.hh"
+
+static logging::logger qlogger("query_ranges_to_vnodes_async");
+
+query_ranges_to_vnodes_generator_async::query_ranges_to_vnodes_generator_async(
+            std::unique_ptr<locator::token_range_splitter> splitter,
+            schema_ptr s,
+            partition_ranges_generator& gen,
+            bool local)
+        : query_ranges_to_vnodes_generator(std::move(splitter), s, {}, local)
+        , _gen(gen)
+        , _gen_exhausted(false) {}
+
+future<std::optional<dht::partition_range_vector>> query_ranges_to_vnodes_generator_async::operator()(size_t n) {
+    if (!_error.empty()) {
```

Comment:
can _gen_exhausted by true upon entry?

Reply:
In this case should a disengaged optional be returned or an engaged optional holding an  empty vector?
Does it matter?